graphql-dxm-provider/src/main/java/org/jahia/modules/graphql/provider/dxm/service/tags/GqlTagQuery.java

```diff
@@ -0,0 +1,118 @@
+ *
+ *     Alternatively, commercial and supported versions of the program - also known as
+ *     Enterprise Distributions - must be used in accordance with the terms and conditions
+ *     contained in a separate written agreement between you and Jahia Solutions Group SA.
+ *
+ *     If you are unsure which license is appropriate for your use,
+ *     please contact the sales department at sales@jahia.com.
+ */
+package org.jahia.modules.graphql.provider.dxm.service.tags;
+
+import graphql.annotations.annotationTypes.GraphQLDescription;
+import graphql.annotations.annotationTypes.GraphQLField;
+import graphql.annotations.annotationTypes.GraphQLName;
+import graphql.annotations.annotationTypes.GraphQLNonNull;
+import graphql.annotations.connection.GraphQLConnection;
+import graphql.schema.DataFetchingEnvironment;
+import org.jahia.modules.graphql.provider.dxm.BaseGqlClientException;
+import org.jahia.modules.graphql.provider.dxm.node.NodeQueryExtensions;
+import org.jahia.modules.graphql.provider.dxm.relay.DXPaginatedDataConnectionFetcher;
+import org.jahia.osgi.BundleUtils;
+import org.jahia.services.content.JCRSessionFactory;
+import org.jahia.services.content.JCRSessionWrapper;
+import org.jahia.services.tags.TaggingService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.jcr.RepositoryException;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+/**
+ * GraphQL root object for Tags related queries
+ */
+@GraphQLName("JCRTags")
+@GraphQLDescription("JCR Queries")
+public class GqlTagQuery {
+
+    private static final Logger logger = LoggerFactory.getLogger(GqlTagQuery.class);
+
+    public GqlTagQuery() {}
+
+    /**
+     * Get GraphQL representations of query for tags suggestions.
+     *
+     * @param prefix The prefix to filter the tags
+     * @param limit Max number of tags to return
+     * @param startPath The root node to start the search
+     * @param environment The execution context
+     * @return GraphQL representations of tags fetched
+     * @throws BaseGqlClientException In case of issues fetching nodes
+     */
+    @GraphQLField
+    @GraphQLName("suggest")
+    @GraphQLDescription("Handles suggestion tags queries")
+    public List<GqlTagNode> suggestTags(
+            @GraphQLName("prefix") @GraphQLNonNull @GraphQLDescription("The prefix of the tags") String prefix,
+            @GraphQLName("limit") @GraphQLNonNull @GraphQLDescription("Max number of tags to get") Long limit,
+            @GraphQLName("startPath") @GraphQLNonNull @GraphQLDescription("The root node to start the search") String startPath,
+            DataFetchingEnvironment environment
+    ) throws RepositoryException {
+
+        try {
+            JCRSessionWrapper jcrSessionWrapper = getSession();
+
+            TaggingService taggingService = BundleUtils.getOsgiService(TaggingService.class, null);
```

Comment:
Could be great to map all the properties hardcoded like: mincount, offset and sortByCount
so we can specified them on the graphQL call if necessary
they could be null, but it can be useful in some cases